### PR TITLE
Initial commit with StackExchange.Redis

### PR DESCRIPTION
**Pull request by alexvaluyskiy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+```hocon
+akka.persistence {
+    journal {
+		plugin = "akka.persistence.journal.redis"
+        redis {
+            # qualified type name of the Redis persistence journal actor
+            class = "Akka.Persistence.Redis.Journal.RedisJournal, Akka.Persistence.Redis"
+
+            #connection string, as described here: https://github.com/StackExchange/StackExchange.Redis/blob/master/Docs/Configuration.md#basic-configuration-strings
+            connection-string = "127.0.0.1:6379"
```
Comment (Horusiath) [inline README.md]: This is not connection string. I think `configuration-string` is more appropriate here.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }    
+
+    snapshot-store {
+		plugin = "akka.persistence.snapshot-store.redis"
+        redis {
+            # qualified type name of the Redis persistence snapshot storage actor
+            class = "Akka.Persistence.Redis.Snapshot.RedisSnapshotStore, Akka.Persistence.Redis"
+
+            #connection string, as described here: https://github.com/StackExchange/StackExchange.Redis/blob/master/Docs/Configuration.md#basic-configuration-strings
+            connection-string = "127.0.0.1:6379"
```
Comment (Horusiath) [inline README.md]: > This is not connection string. I think `configuration-string` is more appropriate here.

File: src/Akka.Persistence.Redis.Tests/RedisJournalSpec.cs [hunk 3]
```diff
@@ -1,35 +1,61 @@
...
+        private static readonly Config SpecConfig;
+        private static readonly string KeyPrefix;
+
+        static RedisJournalSpec()
+        {
+            var connectionString = ConfigurationManager.ConnectionStrings["redis"].ConnectionString;
+            var database = ConfigurationManager.AppSettings["redisDatabase"];
+
+            SpecConfig = ConfigurationFactory.ParseString(@"
+                akka.test.single-expect-default = 50s
```
Comment (Horusiath) [inline src/Akka.Persistence.Redis.Tests/RedisJournalSpec.cs]: Any reason for 50s?

File: src/Akka.Persistence.Redis/Journal/RedisJournal.cs [hunk 4]
```diff
@@ -1,82 +1,120 @@
...
 
-        protected override void PostStop()
+        protected override void PreStart()
         {
-            base.PostStop();
-
-            _pool.Dispose();
+            base.PreStart();
+            _system = Context.System;
+            _database = new Lazy<IDatabase>(() =>
```
Comment (Horusiath) [inline src/Akka.Persistence.Redis/Journal/RedisJournal.cs]: Doesn't database need to be disposed at some point?

File: src/Akka.Persistence.Redis.Tests/RedisJournalSpec.cs (hunk 3 above)
Comment (alexvaluyskiy) [inline src/Akka.Persistence.Redis.Tests/RedisJournalSpec.cs]: Oops

File: src/Akka.Persistence.Redis/Journal/RedisJournal.cs (hunk 4 above)
Comment (alexvaluyskiy) [inline src/Akka.Persistence.Redis/Journal/RedisJournal.cs]: I will fix it

**Comment by alexvaluyskiy** [comment]
@Horusiath why you are creating redisclient on each operation in your implementation?

**Comment by Horusiath** [comment]
If I got it right, the lib I was using was acquiring clients from predefined pool and releasing them once they complete their task.

File: src/Akka.Persistence.Redis/Journal/RedisJournal.cs (hunk 4 above)
Comment (alexvaluyskiy) [inline src/Akka.Persistence.Redis/Journal/RedisJournal.cs]: No. I can't fix it. IDatabase doesn't implement IDisposable. ConnectionMultiplexer does, but I use it inside lazy only

**Comment by alexvaluyskiy** [comment]
Also this PR fixes https://github.com/akkadotnet/Akka.Persistence.Redis/issues/2 and https://github.com/akkadotnet/Akka.Persistence.Redis/issues/3

**Pull request merged**